libsolidity/formal/CHC.cpp

```diff
@@ -994,10 +986,11 @@ void CHC::resetSourceAnalysis()
 	ArraySlicePredicate::reset();
 	m_blockCounter = 0;
 
-	bool usesZ3 = false;
+	/// At this point every enabled solver is available.
+	/// If more than one Horn solver is selected we go with z3.
+	/// We still need the ifdef because of Z3CHCInterface.
 #ifdef HAVE_Z3
```

Comment:
You could move the `#ifdef` into the `if` though. And maybe inside add an assert that `HAVE_Z3` is true. Same with a similar `#ifdef` below.